scripts/code_style_rewrite_branch.py

```diff
@@ -0,0 +1,252 @@
+    if result.returncode != 0:
+        print("Error creating rebase destination branch '"+new_branch+"'.", \
+                file=sys.stderr)
+        return False
+
+    # Rebase the branch atop the given commit
+    result = subprocess.run(["git", "rebase", commit], check=False)
+    if result.returncode != 0:
+        print("Error rebasing branch '"+branch+"' on commit '"+commit+"'.", \
+                file=sys.stderr)
+        delete_branch(new_branch)
+        return False
+
+    return True
+
+def restyle_commit_onto_current(commit_hash: str) -> bool:
+    """
+    Re-style the given commit and add the result as a new commit on the given
+    branch.
+    """
+    #pylint: disable=too-many-return-statements
+    # Get a list of changed files in the commit
+    result = subprocess.run(["git", "diff", "--name-only", "--no-renames", \
+            commit_hash + "~", commit_hash], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        print("Error getting changed files for commit " + commit_hash + \
+                ".", file=sys.stderr)
+        return False
+    changed_files = str(result.stdout, "ascii").strip().split()
+
+    # Get the current HEAD commit
+    result = subprocess.run(["git", "rev-parse", "HEAD"], \
+            stdout=subprocess.PIPE, check=False)
+    if result.returncode != 0:
+        print("Error getting current HEAD commit.", file=sys.stderr)
+        return False
+    head_commit_hash = str(result.stdout, "ascii").strip()
+
+    # Hard reset to the old commit
+    result = subprocess.run(["git", "reset", "--hard", commit_hash], \
+            check=False)
+    if result.returncode != 0:
+        print("Error hard-resetting to commit " + commit_hash + ".", \
+                file=sys.stderr)
+        return False
+
+    # Soft-reset to the current HEAD commit
+    result = subprocess.run(["git", "reset", "--soft", head_commit_hash], \
+            check=False)
+    if result.returncode != 0:
+        print("Error soft-resetting to commit " + head_commit_hash + ".", \
+                file=sys.stderr)
+        return False
+
+    # Unstage files
+    result = subprocess.run(["git", "restore", "--staged", "."], \
+            check=False)
+    if result.returncode != 0:
+        print("Error unstaging files.", file=sys.stderr)
+        return False
+
+    # Filter file list for source files only
+    changed_src_files = list(filter(lambda f: \
+               PurePath(f).match("*.h") \
+            or PurePath(f).match("*.c") \
+            or PurePath(f).match("tests/suites/*.function") \
+            or PurePath(f).match("scripts/data_files/*.fmt"), \
+            changed_files))
+
+    if len(changed_src_files) > 0:
+        # Restyle the source files to the new style
+        if code_style.fix_style(changed_src_files) != 0:
+            return False
+
+    # Add the changed files
+    for changed_file in changed_files:
```

Comment:
Minor: I think this could also be done with a single invocation of git: `["git", "add"] + changed_files`.

Reply:
[Per David](https://github.com/Mbed-TLS/mbedtls/pull/6626#discussion_r1049969836):

> I have stuck to adding files one-by-one as we want to ignore failures due to file-renaming-related issues.

I'm not sure what those issues are. I think David ran into this in a real-world branch but unfortunately I don't know which one.

Reply:
When a file is renamed the list of changed files contains both the new name and the old name. When we try to add the old name it doesn't exist, so the command fails.

I tried to work around this by only adding files that exist but that causes problems with adding deleted files. Ultimately it's simpler to just add one-by-one and ignore failed git-adds.